app/src/pages/playground/PlaygroundOutput.tsx

```diff
@@ -266,7 +266,9 @@ function PlaygroundOutputText(props: PlaygroundInstanceProps) {
         variables: templateVariables,
         language: templateLanguage,
       },
-      tools: instance.tools.map((tool) => tool.definition),
+      tools: instance.tools.length
+        ? instance.tools.map((tool) => tool.definition)
+        : undefined,
```

Comment:
i think server side safety could help too. NBD

Reply:
we also have that. just dumping the raw jsonified input in the input field for the playground span, so this is a little cleaner.